core/src/test/scala/unit/kafka/server/ControllerApisTest.scala

```diff
@@ -523,8 +523,23 @@ class ControllerApisTest {
         new ListPartitionReassignmentsRequestData()).build())))
   }
 
-  @Test
-  def testCreateTopics(): Unit = {
+  object AlwaysExceededControllerMutationQuota extends ControllerMutationQuota {
+    override def isExceeded: Boolean = true
+    override def record(permits: Double): Unit = throw new ThrottlingQuotaExceededException(throttleTime, "quota exceeded in test")
+    override def throttleTime: Int = 1000
+  }
+
+  @ParameterizedTest(name = "testCreateTopics with mutationQuotaExceeded: {0}, validateOnly: {1}")
+  @CsvSource(value = Array(
+    "no,no",
+    "no,yes",
+    "yes,no",
+    "yes,yes"
+  ))
```

Comment:
Instead of `yes/no` for the quota maybe this should be an int so that we can test that we are actually sending the correct partition count. We are requesting 6 potentially-valid partitions, so it might be good to ensure that 5 is not throttled and 6 is throttle.